examples/server/server.cpp

```diff
@@ -3636,7 +3636,7 @@ int main(int argc, char ** argv) {
         server_task task(SERVER_TASK_TYPE_METRICS);
         task.id = ctx_server.queue_tasks.get_new_id();
         ctx_server.queue_results.add_waiting_task_id(task.id);
-        ctx_server.queue_tasks.post(task, true); // high-priority task
+        ctx_server.queue_tasks.post(std::move(task), true); // high-priority task
 
         // get the result
         server_task_result_ptr result = ctx_server.queue_results.recv(task.id);
```

Comment:
The `task` should not be referenced after being moved. We can make this safer like this:

```c++
        const auto id = ctx_server.queue_tasks.get_new_id();

        {
            server_task task(SERVER_TASK_TYPE_METRICS);
            task.id = id;
            ctx_server.queue_results.add_waiting_task_id(task.id);
            ctx_server.queue_tasks.post(std::move(task), true); // high-priority task
        }
```